src/lib/y2network/autoinst/interfaces_reader.rb

```diff
@@ -88,56 +87,81 @@ def load_generic(config, interface_section)
-          )
+          config.ip = load_ipaddr(interface_section)
         end
 
         # handle aliases
-        interface_section.aliases.each_value do |alias_h|
-          ipaddr = IPAddress.from_string(alias_h["IPADDR"])
-          # Assign first netmask, as prefixlen has precedence so it will overwrite it
-          ipaddr.netmask = alias_h["NETMASK"] if alias_h["NETMASK"]
-          ipaddr.prefix = alias_h["PREFIXLEN"].delete("/").to_i if alias_h["PREFIXLEN"]
-          config.ip_aliases << ConnectionConfig::IPConfig.new(ipaddr, label: alias_h["LABEL"])
+        interface_section.aliases.values.each_with_index do |alias_h, index|
+          config.ip_aliases << load_alias(alias_h, id: "_#{index}")
         end
+
+        # startmode
         if !interface_section.startmode.to_s.empty?
           config.startmode = Startmode.create(interface_section.startmode)
         end
         if config.startmode.name == "ifplugd" && !interface_section.ifplugd_priority.to_s.empty?
           config.startmode.priority = interface_section.ifplugd_priority
         end
+
+        # mtu
         config.mtu = interface_section.mtu.to_i if !interface_section.mtu.to_s.empty?
+
+        # ethtool options
         if interface_section.ethtool_options
           config.ethtool_options = interface_section.ethtool_options
         end
+
+        # fw zone
         config.firewall_zone = interface_section.zone if !interface_section.zone.to_s.empty?
+
+        # DHCLIENT_SETHOSTNAME setup
         if !interface_section.dhclient_set_hostname.empty?
           config.dhclient_set_hostname = interface_section.dhclient_set_hostname == "yes"
         end
 
         config
       end
 
+      # Loads and intializates interface_section's ipaddr attribute
+      #
+      # @param section [Hash] hash of AY profile's interface section as obtained from parser
+      #
+      # @return [ConnectionConfig::IPConfig] created ipaddr object
+      def load_ipaddr(section)
+        ipaddr = IPAddress.from_string(section.ipaddr) if !section.ipaddr.empty?
+
+        # Assign first netmask, as prefixlen has precedence so it will overwrite it
+        ipaddr.netmask = section.netmask if !section.netmask.to_s.empty?
+        ipaddr.prefix = section.prefixlen.to_i if !section.prefixlen.to_s.empty?
+
+        broadcast = IPAddress.new(section.broadcast) if !section.broadcast.empty?
+        remote = IPAddress.new(section.remote_ipaddr) if !section.remote_ipaddr.empty?
+
+        ConnectionConfig::IPConfig.new(
+          ipaddr, broadcast: broadcast, remote_address: remote
+        )
+      end
+
+      # Loads and initializates an IP alias according to given hash with alias details
+      #
+      # @param alias_h [Hash] hash of AY profile's alias section as obtained from parser
+      #
+      # @return [ConnectionConfig::IPConfig] alias details
+      def load_alias(alias_h, id: nil)
+        ipaddr = IPAddress.from_string(alias_h["IPADDR"])
```

Comment:
For a moment I thought that the idea of the PR was to check that the alias IPADDR was not empty bu at the end looks like the same code moved to a convenience method, isn't it?

Reply:
no. 

important line is
```
ConnectionConfig::IPConfig.new(ipaddr, id: id, label: alias_h["LABEL"])
```

there is one minor change which makes it work. Before the change it put all aliases under same index in the internal structure (later in ```add_ips``` - if i recall the name correctly - which let t overwriting and we ended always with exactly one ip ... the one under IPADDR or nothing if it was missing)